src/io/fasta.rs

```diff
@@ -70,7 +70,16 @@ impl<R: io::Read> Reader<R> {
         if !self.line.starts_with('>') {
             return Err(io::Error::new(io::ErrorKind::Other, "Expected > at record start."));
         }
-        record.header.push_str(&self.line);
+        record.id = match self.line[1..].trim_right().splitn(2, ' ').nth(0) {
+            Some("") => "".to_owned(),
```

Comment:
You're right, it was redundant. I'm not sure I understand how to handle the None option with map() though. Wouldn't it be something like 
```rust
...nth(0).map(|s| s.to_owned())
```
and if so, what would happen in the None case?